### PR TITLE
[fix/response-transformer] fix for issue #393

### PR DESCRIPTION
**Pull request by shashiranjan84**
Now add will only add header/json if it does not exist,  config `append` added to support adding new value to existing header/json. If header does not exist, a new one will be added. config `replace added to replace the value of existing header/json with new value

### CODE REVIEW

File: kong/plugins/response-transformer/body_transformer.lua [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+function _M.is_json_body(content_type)
+  return content_type and string_find(content_type:lower(), "application/json", nil, true)
+end
+
+function _M.transform_json_body(conf, buffered_data)
+  local json_body = read_json_body(buffered_data)
+  if json_body ~= nil then
+    
+    -- remove key:value to body
+    for _, name, value in iter(conf.remove.json) do
```
Comment (thibaultCha) [inline kong/plugins/response-transformer/body_transformer.lua]: `value` is never used, isn't the linter complaining?

File: kong/plugins/response-transformer/body_transformer.lua [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    if status then
+      return res
+    end
+  end
+end
+
+local function append_value(current_value, value)
+  local current_value_type = type(current_value)
+ 
+  if current_value_type  == "string" then
```
Comment (thibaultCha) [inline kong/plugins/response-transformer/body_transformer.lua]: style: (2 spaces here)

File: kong/plugins/response-transformer/header_transformer.lua [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  end
+end
+
+local function is_json_body(content_type)
+  if content_type then
+    content_type = stringy.strip(content_type):lower()
+    return stringy.startswith(content_type, APPLICATION_JSON)
+  end
+  return false
+end
```
Comment (thibaultCha) [inline kong/plugins/response-transformer/header_transformer.lua]: This function should probably be implemented the same way as the one in body_transformer.

File: kong/plugins/response-transformer/header_transformer.lua [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+---
+--   # Example:
+--   ngx.headers = header_filter.transform_headers(conf, ngx.headers)
+-- We run transformations in following order: remove, replace, add, append. 
+-- @param[type=table] conf Plugin configuration.
+-- @param[type=table] ngx_headers Table of headers, that should be `ngx.headers`
+-- @treturn table A table containing the new headers.
+function _M.transform_headers(conf, ngx_headers)
+  -- remove headers
+  for _, header_name, header_value in iter(conf.remove.headers) do
```
Comment (thibaultCha) [inline kong/plugins/response-transformer/header_transformer.lua]: `header_value` is unused here too, I think the linter will complain.

**Comment by thibaultCha** [comment]
LGTM too

**Comment by shashiranjan84** [comment]
@thibaultCha PR need to be reviewed as I notice a bug in my code. I have made the change and added test.

**Comment by thibaultCha** [comment]
Should have been a new commit to distinguish it

**Pull request merged**